airflow/contrib/hooks/gc_base_hook.py

```diff
@@ -36,6 +39,10 @@ def _authorize(self):
         service_account = connection_extras.get('service_account', False)
         key_path = connection_extras.get('key_path', False)
 
+        kwargs = {}
+        if self.sub:
+            kwargs['sub'] = self.sub
```

Comment:
my thought is this is a bit more extensible if we ever need to add anything else to kwargs.